sound/soc/sof/intel/hda-dai.c

```diff
@@ -422,56 +432,74 @@ static struct snd_soc_cdai_ops sof_probe_compr_ops = {
 struct snd_soc_dai_driver skl_dai[] = {
 {
 	.name = "SSP0 Pin",
+	SND_SOC_STREAM("ssp0 Tx", "ssp0 Rx", 1, 8),
 },
 {
 	.name = "SSP1 Pin",
+	SND_SOC_STREAM("ssp1 Tx", "ssp1 Rx", 1, 8),
 },
 {
 	.name = "SSP2 Pin",
+	SND_SOC_STREAM("ssp2 Tx", "ssp2 Rx", 1, 8),
 },
 {
 	.name = "SSP3 Pin",
+	SND_SOC_STREAM("ssp3 Tx", "ssp3 Rx", 1, 8),
 },
 {
 	.name = "SSP4 Pin",
+	SND_SOC_STREAM("ssp4 Tx", "ssp4 Rx", 1, 8),
 },
 {
 	.name = "SSP5 Pin",
+	SND_SOC_STREAM("ssp5 Tx", "ssp5 Rx", 1, 8),
 },
 {
 	.name = "DMIC01 Pin",
+	SND_SOC_CAPTURE_STREAM("DMIC01 Rx", 2, 8),
 },
 {
 	.name = "DMIC16k Pin",
+	SND_SOC_CAPTURE_STREAM("DMIC16k Rx", 2, 8),
 },
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
 {
 	.name = "iDisp1 Pin",
 	.ops = &hda_link_dai_ops,
+	SND_SOC_PLAYBACK_STREAM("iDisp1 Tx", 1, 32),
 },
 {
 	.name = "iDisp2 Pin",
 	.ops = &hda_link_dai_ops,
+	SND_SOC_PLAYBACK_STREAM("iDisp2 Tx", 1, 32),
 },
 {
 	.name = "iDisp3 Pin",
 	.ops = &hda_link_dai_ops,
+	SND_SOC_PLAYBACK_STREAM("iDisp3 Tx", 1, 32),
 },
 {
 	.name = "iDisp4 Pin",
 	.ops = &hda_link_dai_ops,
+	SND_SOC_PLAYBACK_STREAM("iDisp4 Tx", 1, 32),
```

Comment:
hold on this one while we verify. I don't see how we can exceed d the 16 channel DMA limit.

Reply:
the answer is 8 channels for all iDISP links.